daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -428,6 +442,136 @@ execClean projectOpts = do
             removeAndWarn "dist"
             putStrLn "Removed build artifacts."
 
+
+
+data Action = ACreate (LF.Qualified LF.TypeConName) 
+            | AExercise (LF.Qualified LF.TypeConName) LF.ChoiceName deriving (Eq, Ord, Show )
+
+
+startFromUpdate :: DS.Set (LF.Qualified LF.ExprValName) -> LF.World -> LF.Update -> DS.Set Action
+startFromUpdate seen world update = case update of 
```

Comment:
You might want to use the recursion-schemes variant of LF.Update for this, i.e., `UpdateF` given that we already have that or as @neil-da suggested uniplate.